src/components/select-field/Overlay.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+import * as React from 'react';
+import getProp from 'lodash/get';
+import classNames from 'classnames';
+import { getDimensions, getViewportIntersections, type ElementDimensions } from '../../utils/dom';
+import { VIEWPORT_BORDERS } from '../../constants';
+
+import './Overlay.scss';
+
+type Props = {
+    children: React.Node,
+    className?: string,
+    isOpen?: boolean,
+};
+
+type Translations = {
+    x?: number,
+    y?: number,
+};
+class Overlay extends React.Component<Props, State> {
+    overlayRef: ?HTMLDivElement;
+
+    buttonRef: ?HTMLButtonElement;
+
+    constructor(props) {
+        super(props);
+        this.buttonRef = React.createRef();
+        this.overlayRef = React.createRef();
+    }
+
+    getOverlayContentDimensions = (): ElementDimensions => {
+        const node = this.overlayRef.current;
+        const content = node && node.firstChild;
+
+        const dimensions = getDimensions(content);
+
+        return dimensions;
+    };
+
+    getOverlayIntersections = (): Array<string> => {
+        const { isOpen } = this.props;
+
+        if (!isOpen) {
+            return [];
+        }
+
+        const dimensions = this.getOverlayContentDimensions();
+        const node = this.overlayRef.current;
+        const boundingClientRect = node && node.getBoundingClientRect();
+        const bounding = {
+            top: boundingClientRect.top,
+            left: boundingClientRect.left,
+            bottom: boundingClientRect.top + dimensions.height,
+            right: boundingClientRect.left + dimensions.width,
+        };
+
+        return getViewportIntersections(bounding);
+    };
+
+    getOverlayTranslations = (intersections: Array<string>): Translations => {
+        const { height: overlayHeight, width: overlayWidth } = this.getOverlayContentDimensions();
+        const { height: buttonHeight, width: buttonWidth } = this.buttonRef.current.getBoundingClientRect();
```

Comment:
Do we need to check the existence of `buttonRef` just like checking `overlayRef` above?